README.md
Magpie
------

Magpie contains a number of scripts for running Big Data software in
HPC environments.  Thus far, Hadoop, Spark, Hbase, Storm, Pig,
Phoenix, Kafka, Zeppelin, Zookeeper, and Alluxio are supported. It
currently supports running over the parallel file system Lustre and
running over any generic network filesytem.  There is
scheduler/resource manager support for Slurm, Moab, Torque, and LSF.

Some of the features presently supported:

- Run jobs interactively or via scripts.
- Run against a number of filesystem options, such as HDFS, HDFS over
  Lustre, HDFS over a generic network filesystem, Lustre directly, or
  a generic network filesystem.
- Take advantage of SSDs/NVRAM for local caching if available
- Make decent optimizations for your hardware

Experimental support for several distributed machine learning
frameworks has also been added.  Presently tensorflow, tensorflow
w/ horovod, and ray is supported.

Basic Idea
----------

The basic idea behind these scripts are to:

1) Submit a Magpie batch script to allocate nodes on a cluster using
   your HPC scheduler/resource manager.  Slurm, Slurm+mpirun,
   Moab+Slurm, Moab+Torque and LSF+mpirun are currently supported.

2) The batch script will create configuration files for all
   appropriate projects (Hadoop, Spark, etc.)  The configuration files
   will be setup so the rank 0 node is the "master".  All compute
   nodes will have configuration files created that point to the node
   designated as the master server.

   The configuration files will be populated with values for your
   filesystem choice and the hardware that exists in your cluster.
   Reasonable attempts are made to determine optimal values for your
   system and hardware (they are almost certainly better than the
   default values).  A number of options exist in the batch scripts to
   adjust these values for individual jobs.

3) Launch daemons on all nodes.  The rank 0 node will run master
   daemons, such as the Hadoop Namenode.  All remaining nodes will run
   appropriate worker daemons, such as the Hadoop Datanodes.

4) Now you have a mini big data cluster to do whatever you want.  You
   can log into the master node and interact with your mini big data
   cluster however you want.  Or you could have Magpie run a script to
   execute your big data calculation instead.

5) When your job completes or your allocation time has run out, Magpie
   will cleanup your job by tearing down daemons.  When appropriate,
   Magpie may also do some additional cleanup work to hopefully make
   re-execution on later runs cleaner and faster.

Supported Packages & Versions
-----------------------------

For a complete list of supported package versions and dependencies,
please see ```doc/README```.  The following can be considered a
summary of support.

Hadoop - 2.2.0, 2.3.0, 2.4.X, 2.5.X, 2.6.X, 2.7.X, 2.8.X, 2.9.X,
         3.0.X, 3.1.X, 3.2.X

Spark - 1.1.X, 1.2.X, 1.3.X, 1.4.X, 1.5.X, 1.6.X, 2.0.X, 2.1.X, 2.2.X,
        2.3.X, 2.4.X, 3.0.X

Hbase - 1.0.X, 1.1.X, 1.2.X, 1.3.X, 1.4.X, 1.5.X, 1.6.X

Hive - 2.3.0

Pig - 0.13.0, 0.14.0, 0.15.0, 0.16.0, 0.17.0

Zookeeper - 3.4.X

Storm - 0.9.X, 0.10.X, 1.0.X, 1.1.X, 1.2.X

Phoenix - 4.5.X, 4.6.0, 4.7.0, 4.8.X, 4.9.0, 4.10.1, 4.11.0, 4.12.0,
          4.13.X, 4.14.0

Kafka - 2.11-0.9.0.0

Zeppelin - 0.6.X, 0.7.X, 0.8.X

Alluxio - 2.3.0

TensorFlow - 1.9, 1.12

Ray - 0.7.0

Older Supported Packages & Features
-----------------------------------

Some packages and features were dropped due to lack of interest, the
software becoming old/deprecated, and/or their initial experimental
addition into Magpie.  If you are interested in them, please look at
older versions for supported versions and documentation.  If you are
very interested in support in current versions of Magpie beyond an
experimental nature, please submit a support request and we can
reconsider adding it back in.

Removed in Magpie 2.0

   - Hadoop 1.X support
   - Tachyon
   - UDA/uda-plugin for Hadoop
   - HDFS Federation in Hadoop
   - IntelLustre option for a Hadoop Filesystem
   - MagpieNetworkFS option for a Hadoop Filesystem

Removed in Magpie 3.0

   - Spark 0.9.X support
   - Hbase 0.98.X and 0.99.X support
   - Mahout

Documentation
-------------

All documentation is in the 'doc' subdirectory.  Please see the
doc/README file as a starting point.  It provides general instructions
as well as pointers to documentation for each project, setup
requirements, ability to do local configurations, tips & tricks, and
more information.

Release
-------

Magpie is release under a GPL license. For more information, see the [COPYING](/COPYING) file.

`LLNL-CODE-644248`


NEWS
Magpie 3.0
----------

Magpie 3.0 includes some new software support, but primarily removes a
lot of older support.

Legacy support Removed
- Spark 0.9.X support
- Hbase 0.98.X and 0.99.X support
- Mahout

Backwards Compatibility Notes
- Rename SPARK_SLAVE_COUNT to SPARK_WORKER_COUNT
- Rename SPARK_SLAVE_CORE_COUNT to SPARK_WORKER_CORE_COUNT
- Remove legacy support for HADOOP_SLAVE_COUNT and
  HADOOP_SLAVE_CORE_COUNT.  HADOOP_WORKER_COUNT and
  HADOOP_WORKER_CORE_COUNT should be used.
- SPARK_JOB_CLASSPATH, SPARK_JOB_LIBRARY_PATH, and SPARK_JOB_JAVA_OPTS
  environment variables are removed.  Users should use
  --driver-class-path, --driver-library-path, and
  --driver-java-options in spark-submit instead.
- Remove legacy support of STORM_WORKERS_COUNT (plural), users should
  use STORM_WORKER_COUNT.

Magpie 2.5
----------
Add Alluxio support.
Support Spark 3.0.0-bin-hadoop2.7, 3.0.0-bin-hadoop3.2,
  Spark 2.4.6-bin-hadoop2.6, 2.4.6-bin-hadoop2.7
- Update Spark default to 3.0.0-bin-hadoop3.2
Support Hbase 1.2.7, 1.3.3, 1.3.4, 1.3.5, 1.5.0, 1.6.0.
- Update Hbase default to 1.6.0

Magpie 2.4
----------
Support Hadoop 3.1.3, 3.2.1
- Update Hadoop default to 3.2.1
Add support for Storm 1.2.3.
- Update default Storm to 1.2.3.
Add support for Hbase 1.4.10, 1.4.13
- Update default Hbase to 1.4.10, 1.4.13
Add support for Zeppelin 0.8.2.
- Update default Zeppelin to 0.8.2.
Support Spark 2.3.3-bin-hadoop2.6, 2.3.3-bin-hadoop2.7, Spark
 2.3.4-bin-hadoop2.6, 2.3.4-bin-hadoop2.7, Spark 2.4.5-bin-hadoop2.6,
 2.4.5-bin-hadoop2.7
- Update Spark default to 2.4.4-bin-hadoop2.7

Magpie 2.3
----------
Add experimental support Ray.
Support Spark 2.4.2-bin-hadoop2.6, 2.4.2-bin-hadoop2.7,
  2.4.3-bin-hadoop2.6, 2.4.3-bin-hadoop2.7
- Update Spark default to 2.4.3-bin-hadoop2.7
Add support for Zookeeper 3.4.14.
- Update default Zookeeper to 3.4.14.

Magpie 2.2
----------
Support SPARK_IO_COMPRESSION_CODEC in Spark.
With sbatch & srun, add experimental support for distributed
 tensorflow support.
Add new submission script type "sbatchmpirun"
With sbatch & mpirun, add experimental support for Tensorflow with Horovod
Support new MAGPIE_PRE_EXECUTION_RUN and MAGPIE_POST_EXECUTION_RUN
 options.  These options are similar to MAGPIE_PRE_JOB_RUN and
 MAGPIE_POST_JOB_RUN, but run before/after the entire software stack
 has been brought up / torn down.
Support Spark 2.4.1-bin-hadoop2.6, 2.4.1-bin-hadoop2.7
- Update Spark default to 2.4.1-bin-hadoop2.7

Magpie 2.1
----------
Support SPARK_NETWORK_TIMEOUT configuration option.
Support SPARK_RDD_COMPRESS configuration option.
Fix various deprecated warnings.
Fix broken Zeppelin dependencies.
Alter default Spark webui ports to avoid conflicts with other web services.
Support Hbase 1.4.7, 1.4.8, 1.4.9
- Update Hbase default to 1.4.9
Support Spark 2.4.0-bin-hadoop2.6, 2.4.0-bin-hadoop2.7, Spark
  2.3.2-bin-hadoop2.6, 2.3.2-bin-hadoop2.7
- Update Spark default to 2.4.0-bin-hadoop2.7
Support Hadoop 2.7.6, 2.7.7, 2.8.5, 2.9.2, 3.1.2, 3.2.0
- Update Hadoop default to 3.2.0
Support Zeppelin 0.8.0, 0.8.1
- Update Zeppelin default to 0.8.1
- Implement Shiro basic auth for Zeppelin, default requires a password being set

Magpie 2.0
----------

Magpie 2.0 includes new software support, but primarily involves a
large amount of re-architecture and removal of legacy experimental
support.  Job submission scripts from Magpie 1.X versions will be
incompatible with Magpie 2.0.  So please read details below for
highlights.

New Support

o Support Hadoop version 3

  - Support new HADOOP_WORKER_COUNT & HADOOP_WORKER_CORE_COUNT
    environment variables.

  - Update default Hadoop version to 3.1.1.

o Support Hive, initially with Hive v2.3.0

Re-architecture and changes

o Rearchitect MAGPIE_JOB_TYPE to be overarching job configuration
  variable.

  - "setuponly" and "interactive" jobs are now configured by
    MAGPIE_JOB_TYPE and not through any of the former "MODE" variables
    (e.g. HADOOP_MODE, SPARK_MODE, etc.)

  - Generic "script" jobs are now configured by MAGPIE_JOB_TYPE and
    not through any of the former "MODE" variables.
    (e.g. HADOOP_MODE, SPARK_MODE, etc.)

    However, "script" is still available for those projects that run a
    script through a specific service, e.g. PIG and PHOENIX.

o Rearchitect HADOOP_SETUP_TYPE to be version agnostic.  It now takes
  "MR", "YARN", or "HDFS" as inputs.

  - A new submission script variant to support Spark w/ Yarn and w/o
    HDFS is now available.

o "MODE" variables (e.g. HADOOP_MODE, SPARK_MODE, etc.) are now named
  "JOB" (e.g. HADOOP_JOB, SPARK_JOB, etc.).  This change was done to
  indicate that they truly only run jobs and are not meant for
  configuring the project in any way.

  - As a result the following configurations have been removed.

    "launch" from HADOOP_JOB
    "hdfsonly" from HADOOP_JOB
    "server" from KAFKA_JOB
    "server" from PHOENIX_JOB

testsuite/test-common.sh
#!/bin/bash

java16="1.6"
java17="1.7"
java18="1.8"

hadoop_decomissionhdfs_minimum="2.3.0"

hadoop2Xjava16versions="2.2.0 2.3.0 2.4.0 2.4.1 2.5.0 2.5.1 2.5.2"
hadoop2Xjava16versions_javaversion=${java16}
hadoop2Xjava17versions="2.6.0 2.6.1 2.6.2 2.6.3 2.6.4 2.6.5 2.7.0 2.7.1 2.7.2 2.7.3 2.8.0 2.8.1 2.8.2 2.8.3 2.8.4 2.9.0 2.9.1"
hadoop2Xjava17versions_javaversion=${java17}
hadoop2Xjava18versions="2.7.4 2.7.5 2.7.6 2.7.7 2.8.5 2.9.2"
hadoop2Xjava18versions_javaversion=${java18}
hadoop3Xjava18versions="3.0.0 3.0.1 3.0.2 3.0.3 3.1.0 3.1.1 3.1.2 3.1.3 3.2.0 3.2.1"
hadoop3Xjava18versions_javaversion=${java18}

hadoop_test_groups="hadoop2Xjava16versions hadoop2Xjava17versions hadoop2Xjava18versions hadoop3Xjava18versions"
hadoop_all_versions="${hadoop2Xjava16versions} ${hadoop2Xjava17versions} ${hadoop2Xjava18versions} ${hadoop3Xjava18versions}"

pighadoop26java16versions="0.13.0 0.14.0"
pighadoop26java16versions_hadoopversion="2.6.0"
pighadoop26java16versions_javaversion=${java16}
pighadoop27java17versions="0.15.0 0.16.0 0.17.0"
pighadoop27java17versions_hadoopversion="2.7.0"
pighadoop27java17versions_javaversion=${java17}

pig_test_groups="pighadoop26java16versions pighadoop27java17versions"
pig_all_versions="${pighadoop26java16versions} ${pighadoop27java17versions}"

hbasehadoop27zookeeper34java17versions="1.0.0 1.0.1 1.0.1.1 1.0.2 1.0.3 1.1.0 1.1.0.1 1.1.1 1.1.2 1.1.3 1.1.4 1.1.5 1.1.6 1.1.7 1.1.8 1.1.9 1.1.10 1.1.11 1.1.12 1.1.13 1.2.0 1.2.1 1.2.2 1.2.3 1.2.4 1.2.5 1.2.6 1.2.6.1 1.3.0 1.3.1 1.3.2 1.3.2.1 1.4.0 1.4.1 1.4.2 1.4.3 1.4.4 1.4.5 1.4.6 1.4.7 1.4.8 1.4.9"
hbasehadoop27zookeeper34java17versions_hadoopversion="2.7.0"
hbasehadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
hbasehadoop27zookeeper34java17versions_javaversion=${java17}
hbasehadoop27zookeeper34java18versions="1.2.7 1.3.3 1.3.4 1.3.5 1.4.10 1.4.13 1.5.0 1.6.0"
hbasehadoop27zookeeper34java18versions_hadoopversion="2.7.0"
hbasehadoop27zookeeper34java18versions_zookeeperversion="3.4.14"
hbasehadoop27zookeeper34java18versions_javaversion=${java18}

hbase_test_groups="hbasehadoop27zookeeper34java17versions hbasehadoop27zookeeper34java18versions"
hbase_all_versions="${hbasehadoop27zookeeper34java17versions} ${hbasehadoop27zookeeper34java18versions}"

hivehadoop27zookeeper34java17versions="2.3.0"
hivehadoop27zookeeper34java17versions_hadoopversion="2.7.0"
hivehadoop27zookeeper34java17versions_zookeeperversion="3.4.11"
hivehadoop27zookeeper34java17versions_javaversion=${java17}

hive_test_groups="hivehadoop27zookeeper34java17versions"
hive_all_versions="${hivehadoop27zookeeper34java17versions}"

phoenixhbase10hadoop27zookeeper34java17versions="4.5.0-HBase-1.0 4.5.1-HBase-1.0 4.5.2-HBase-1.0 4.6.0-HBase-1.0 4.7.0-HBase-1.0 4.8.0-HBase-1.0 4.8.1-HBase-1.0 4.8.2-HBase-1.0"
phoenixhbase10hadoop27zookeeper34java17versions_hbaseversion="1.0.0"
phoenixhbase10hadoop27zookeeper34java17versions_hadoopversion="2.7.0"
phoenixhbase10hadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
phoenixhbase10hadoop27zookeeper34java17versions_javaversion=${java17}

phoenixhbase11hadoop27zookeeper34java17versions="4.5.0-HBase-1.1 4.5.1-HBase-1.1 4.5.2-HBase-1.1 4.6.0-HBase-1.1 4.7.0-HBase-1.1 4.8.0-HBase-1.1 4.8.1-HBase-1.1 4.8.2-HBase-1.1 4.9.0-HBase-1.1 4.10.0-HBase-1.1 4.11.0-HBase-1.1 4.12.0-HBase-1.1 4.13.1-HBase-1.1 4.14.0-HBase-1.1"
phoenixhbase11hadoop27zookeeper34java17versions_hbaseversion="1.1.0"
phoenixhbase11hadoop27zookeeper34java17versions_hadoopversion="2.7.0"
phoenixhbase11hadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
phoenixhbase11hadoop27zookeeper34java17versions_javaversion=${java17}

phoenixhbase12hadoop27zookeeper34java17versions="4.8.0-HBase-1.2 4.8.1-HBase-1.2 4.8.2-HBase-1.2 4.9.0-HBase-1.2 4.10.0-HBase-1.2 4.11.0-HBase-1.2 4.12.0-HBase-1.2 4.13.1-HBase-1.2 4.14.0-HBase-1.2"
phoenixhbase12hadoop27zookeeper34java17versions_hbaseversion="1.2.0"
phoenixhbase12hadoop27zookeeper34java17versions_hadoopversion="2.7.0"
phoenixhbase12hadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
phoenixhbase12hadoop27zookeeper34java17versions_javaversion=${java17}

phoenixhbase13hadoop27zookeeper34java17versions="4.11.0-HBase-1.3 4.12.0-HBase-1.3 4.13.0-HBase-1.3 4.13.1-HBase-1.3 4.14.0-HBase-1.3"
phoenixhbase13hadoop27zookeeper34java17versions_hbaseversion="1.3.0"
phoenixhbase13hadoop27zookeeper34java17versions_hadoopversion="2.7.0"
phoenixhbase13hadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
phoenixhbase13hadoop27zookeeper34java17versions_javaversion=${java17}

phoenixhbase14hadoop27zookeeper34java17versions="4.14.0-HBase-1.4"
phoenixhbase14hadoop27zookeeper34java17versions_hbaseversion="1.4.0"
phoenixhbase14hadoop27zookeeper34java17versions_hadoopversion="2.7.0"
phoenixhbase14hadoop27zookeeper34java17versions_zookeeperversion="3.4.14"
phoenixhbase14hadoop27zookeeper34java17versions_javaversion=${java17}

phoenix_test_groups="phoenixhbase10hadoop27zookeeper34java17versions phoenixhbase11hadoop27zookeeper34java17versions phoenixhbase12hadoop27zookeeper34java17versions phoenixhbase13hadoop27zookeeper34java17versions phoenixhbase14hadoop27zookeeper34java17versions"
phoenix_all_versions="${phoenixhbase10hadoop27zookeeper34java17versions} ${phoenixhbase11hadoop27zookeeper34java17versions} ${phoenixhbase12hadoop27zookeeper34java17versions} ${phoenixhbase13hadoop27zookeeper34java17versions} ${phoenixhbase14hadoop27zookeeper34java17versions}"

spark1Xjava16hadoop23versions="1.1.0-bin-hadoop2.3 1.1.1-bin-hadoop2.3 1.2.0-bin-hadoop2.3 1.2.1-bin-hadoop2.3 1.2.2-bin-hadoop2.3 1.3.0-bin-hadoop2.3 1.3.1-bin-hadoop2.3 1.4.0-bin-hadoop2.3 1.4.1-bin-hadoop2.3"
spark1Xjava16hadoop23versions_hadoopversion="2.3.0"
spark1Xjava16hadoop23versions_javaversion=${java16}
spark1Xjava16hadoop24versions="1.1.0-bin-hadoop2.4 1.1.1-bin-hadoop2.4 1.2.0-bin-hadoop2.4 1.2.1-bin-hadoop2.4 1.2.2-bin-hadoop2.4 1.3.0-bin-hadoop2.4 1.3.1-bin-hadoop2.4 1.4.0-bin-hadoop2.4 1.4.1-bin-hadoop2.4"
spark1Xjava16hadoop24versions_hadoopversion="2.4.0"
spark1Xjava16hadoop24versions_javaversion=${java16}
spark1Xjava17hadoop26versions="1.3.1-bin-hadoop2.6 1.4.0-bin-hadoop2.6 1.4.1-bin-hadoop2.6 1.5.0-bin-hadoop2.6 1.5.1-bin-hadoop2.6 1.5.2-bin-hadoop2.6 1.6.0-bin-hadoop2.6 1.6.1-bin-hadoop2.6 1.6.2-bin-hadoop2.6 1.6.3-bin-hadoop2.6"
spark1Xjava17hadoop26versions_hadoopversion="2.6.0"
spark1Xjava17hadoop26versions_javaversion=${java17}
spark2Xjava17hadoop26versions="2.0.0-bin-hadoop2.6 2.0.1-bin-hadoop2.6 2.0.2-bin-hadoop2.6 2.1.0-bin-hadoop2.6 2.1.1-bin-hadoop2.6 2.1.2-bin-hadoop2.6"
spark2Xjava17hadoop26versions_hadoopversion="2.6.0"
spark2Xjava17hadoop26versions_javaversion=${java17}
spark2Xjava17hadoop27versions="2.0.0-bin-hadoop2.7 2.0.1-bin-hadoop2.7 2.0.2-bin-hadoop2.7 2.1.0-bin-hadoop2.7 2.1.1-bin-hadoop2.7 2.1.2-bin-hadoop2.7"
spark2Xjava17hadoop27versions_hadoopversion="2.7.0"
spark2Xjava17hadoop27versions_javaversion=${java17}
spark2Xjava18hadoop26versions="2.2.0-bin-hadoop2.6 2.2.1-bin-hadoop2.6 2.3.0-bin-hadoop2.6 2.3.1-bin-hadoop2.6 2.3.2-bin-hadoop2.6 2.3.3-bin-hadoop2.6 2.3.4-bin-hadoop2.6 2.4.0-bin-hadoop2.6 2.4.1-bin-hadoop2.6 2.4.2-bin-hadoop2.6 2.4.3-bin-hadoop2.6 2.4.4-bin-hadoop2.6 2.4.5-bin-hadoop2.6 2.4.6-bin-hadoop2.6"
spark2Xjava18hadoop26versions_hadoopversion="2.6.0"
spark2Xjava18hadoop26versions_javaversion=${java18}
spark2Xjava18hadoop27versions="2.2.0-bin-hadoop2.7 2.2.1-bin-hadoop2.7 2.3.0-bin-hadoop2.7 2.3.1-bin-hadoop2.7 2.3.2-bin-hadoop2.7 2.3.3-bin-hadoop2.7 2.3.4-bin-hadoop2.7 2.4.0-bin-hadoop2.7 2.4.1-bin-hadoop2.7 2.4.2-bin-hadoop2.7 2.4.3-bin-hadoop2.7 2.4.4-bin-hadoop2.7 2.4.5-bin-hadoop2.7 2.4.6-bin-hadoop2.7 3.0.0-bin-hadoop2.7"
spark2Xjava18hadoop27versions_hadoopversion="2.7.0"
spark2Xjava18hadoop27versions_javaversion=${java18}
spark3Xjava18hadoop32versions="3.0.0-bin-hadoop3.2"
spark3Xjava18hadoop32versions_hadoopversion="3.2.0"
spark3Xjava18hadoop32versions_javaversion=${java18}

spark_test_groups="spark1Xjava16hadoop23versions spark1Xjava16hadoop24versions spark1Xjava17hadoop26versions spark2Xjava17hadoop26versions spark2Xjava17hadoop27versions spark2Xjava18hadoop26versions spark2Xjava18hadoop27versions spark3Xjava18hadoop32versions"
spark_all_versions="${spark1Xjava16hadoop23versions} ${spark1Xjava16hadoop24versions} ${spark1Xjava17hadoop26versions} ${spark2Xjava17hadoop26versions} ${spark2Xjava17hadoop27versions} ${spark2Xjava18hadoop26versions} ${spark2Xjava18hadoop27versions} ${spark3Xjava18hadoop32versions}"

stormzookeeper34java16versions="0.9.3 0.9.4"
stormzookeeper34java16versions_zookeeperversion="3.4.6"
stormzookeeper34java16versions_javaversion=${java16}
stormzookeeper34java17versions="0.9.5 0.9.6 0.9.7 0.10.0 0.10.1 0.10.2 1.0.0 1.0.1 1.0.2 1.0.3 1.0.4 1.1.0 1.1.1 1.1.2 1.2.0 1.2.1 1.2.2"
stormzookeeper34java17versions_zookeeperversion="3.4.14"
stormzookeeper34java17versions_javaversion=${java17}
stormzookeeper34java18versions="1.2.3"
stormzookeeper34java18versions_zookeeperversion="3.4.14"
stormzookeeper34java18versions_javaversion=${java18}

storm_test_groups="stormzookeeper34java16versions stormzookeeper34java17versions stormzookeeper34java18versions"
storm_all_versions="${stormzookeeper34java16versions} ${stormzookeeper34java17versions} ${stormzookeeper34java18versions}"

kafkazookeeper34java17versions="2.11-0.9.0.0"
kafkazookeeper34java17versions_zookeeperversion="3.4.14"
kafkazookeeper34java17versions_javaversion=${java17}

kafka_test_groups="kafkazookeeper34java17versions"
kafka_all_versions="${kafkazookeeper34java17versions}"

zookeeperjava17versions="3.4.0 3.4.1 3.4.2 3.4.3 3.4.4 3.4.5 3.4.6 3.4.7 3.4.8 3.4.9 3.4.10 3.4.11 3.4.12 3.4.13"
zookeeperjava17versions_javaversion=${java17}
zookeeperjava18versions="3.4.14"
zookeeperjava18versions_javaversion=${java18}

zookeeper_test_groups="zookeeperjava17versions zookeeperjava18versions"
zookeeper_all_versions="${zookeeperjava17versions} ${zookeeperjava18versions}"

zeppelinspark16java17versions="0.6.0 0.6.1 0.6.2 0.7.0 0.7.1 0.7.2 0.7.3"
zeppelinspark16java17versions_sparkversion="1.6.0-bin-hadoop2.6"
zeppelinspark16java17versions_javaversion="${java17}"
zeppelinspark16java18versions="0.8.0 0.8.1 0.8.2"
zeppelinspark16java18versions_sparkversion="1.6.0-bin-hadoop2.6"
zeppelinspark16java18versions_javaversion="${java18}"

zeppelin_test_groups="zeppelinspark16java17versions zeppelinspark16java18versions"
zeppelin_all_versions="${zeppelinspark16java17versions} ${zeppelinspark16java18versions}"

alluxiojava18versions="2.3.0"
alluxiojava18versions_javaversion=${java18}

alluxio_test_groups="alluxiojava18versions"
alluxio_all_versions="${alluxiojava18versions}"
